dataplane/ebpf/src/egress/icmp.rs

```diff
@@ -0,0 +1,93 @@
+use core::mem;
+
+use aya_bpf::{
+    bindings::TC_ACT_PIPE,
+    helpers::bpf_csum_diff,
+    programs::TcContext,
+};
+use aya_log_ebpf::info;
+
+use crate::{
+    bindings::{iphdr, icmphdr},
+    utils::{csum_fold_helper, ip_from_int, ptr_at, ETH_HDR_LEN, IP_HDR_LEN},
+    CONNTRACK,
+};
+
+const ICMP_HDR_LEN: usize = mem::size_of::<icmphdr>();
+const ICMP_PROTO_TYPE_UNREACH: u8 = 3;
+
+pub fn handle_icmp_egress(ctx: TcContext) -> Result<i32, i64> {
+    let ip_hdr: *mut iphdr = unsafe { ptr_at(&ctx, ETH_HDR_LEN) }?;
+
+    let icmp_header_offset = ETH_HDR_LEN + IP_HDR_LEN;
+
+    let icmp_hdr: *mut icmphdr = unsafe { 
+        ptr_at(
+            &ctx,
+            icmp_header_offset
+        )?
+    };
+
+    // We only care about redirecting port unreachable messages currently so a 
+    // UDP client can tell when the server is shutdown
+    if unsafe { (*icmp_hdr).type_ } != ICMP_PROTO_TYPE_UNREACH { 
+        return Ok(TC_ACT_PIPE);
+    }   
+
+    let dest_addr = unsafe { (*ip_hdr).daddr };
+    
+    let new_src = unsafe { CONNTRACK.get(&dest_addr) }.ok_or(TC_ACT_PIPE)?;
```

Comment:
I almost wonder if it would be worth naming our "conntrack" something more unique to us, to avoid any confusion that this is our own implementation. What do you think about `BLIXT_CONNTRACK`?

Reply:
I'll do it :)